src/coord/src/coord.rs

```diff
@@ -951,9 +962,18 @@ impl Coordinator {
     }
 
     /// Updates the upper frontier of a named view.
-    fn update_upper(&mut self, name: &GlobalId, mut changes: ChangeBatch<Timestamp>) {
-        if let Some(index_state) = self.indexes.get_mut(name) {
-            let changes: Vec<_> = index_state.upper.update_iter(changes.drain()).collect();
+    fn update_upper(
+        &mut self,
+        mut feedback: FrontierFeedback,
+        durability_updates: &mut Vec<(GlobalId, Antichain<Timestamp>)>,
+    ) {
```

Comment:
It looks like frontier updates are being woven together with something about durability updates. Either that needs a whole lot of explaining somewhere, or ideally instead we could break apart the behavior to be more clear about what things are going on. The documentation for the method hasn't changed for example, which can't be right.